include/utils/filesystem.hpp

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef UTILS_FILESYSTEM_HPP
+#define UTILS_FILESYSTEM_HPP
+
+#include <string>
+
+#include <aos/common/tools/error.hpp>
+
+namespace aos::common::utils {
+/*
+ * Creates a temporary directory.
+ *
+ * @param dir directory path.
+ * @param pattern directory name pattern.
```

Comment:
please add more comments how the pattern look like

Reply:
Done